### DISCUSSION TITLE
Get user email from GitHub action

### DISCUSSION THREAD

**Discussion by alienxp03** [community/community#40405] [Actions]
### Select Topic Area

Product Feedback

### Body

In some of our workflow, certain steps require us to create a new commit and push it to our repo. Currently, we could use `github.actor` to get the user GitHub's username, however, there's no official way to retrieve the user's email. We can commit without an email, however we prefer not to. There is various workaround but each has its own flaws. `github.event.pusher.email` is only available for push events. For `workflow_dispatch` event, the value will be empty.

Is there any reason why GitHub doesn't expose the user's email in the [GitHub Action Context](https://docs.github.com/en/actions/learn-github-actions/contexts)? Would be great if we can retrieve the user's username & email based on who triggered the event (_pull request, new commits, workflow_dispatch_).

**Comment by airtower-luna** [comment]
You generally shouldn't try to use the user's email address as the author email of a commit they didn't make. The [`actions/checkout` documentation](https://github.com/actions/checkout#push-a-commit-using-the-built-in-token) recommends a mail address that should make the commit visible as having been created by an Actions workflow.

**Comment by mimiandcanada** [comment]
I see that there is https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request and under user object, there is email and name, but they are not populated. Anyone know how these are populated?

**Comment by shqear93** [comment]
any update regarding this?!

**Comment by ovidb** [comment]
I'm also interested in a an official way of doing this.

We have a use-case where this is required. For us the result of the action is a PR raised on the behalf of the user (we want to simplify some of the complexity around correct messages and branching strategy) 

In our case is a must to have the user's username and email simply because otherwise they will be able to approve their own PR 😅

**Comment by ghost** [comment]
> Is there any reason why GitHub doesn't expose the user's email in the [GitHub Action Context](https://docs.github.com/en/actions/learn-github-actions/contexts)? Would be great if we can retrieve the user's username & email based on who triggered the event (pull request, new commits, workflow_dispatch).

second this..  please make the user's email available in the context

**Comment by AAriam** [comment]
You can obtain the default `@users.noreply.github.com` email of the actor from the event payload, contained in the `github` context. 

The default email address has the following template: `{USER-ID}+{USERNAME}@users.noreply.github.com`. Both the user-ID and username can be obtained from the event payload and used to construct the default email address.

The user-ID is available at `github.event.sender.id` and the username at `github.event.sender.login`. According to GitHub documentation, the `sender` property is included in every payload.

**Reply by reitzig** [reply]
Since `github.actor_id` exists ([ref](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)), this should work: 

```bash
git config user.name "${{ github.actor }}"
git config user.email "${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com"
```

**Reply by TWiStErRob** [reply]
Relevant docs: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#about-commit-email-addresses
